tevfik_aksoy/main.cpp

```diff
@@ -120,28 +154,55 @@ void write_to_file(const GraphPrinter& graph_printer,
   jsonfile.close();
 }
 
+void prepare_temp_directory() {
+  std::filesystem::create_directory("./temp");
+}
+
+void prepare_log_file(Logger& logger) {
+  logger.set_file("./temp/log.txt");
+}
+
 int main() {
   const int threads_count = handle_threads_count_input();
   const int graphs_count = handle_graphs_count_input();
   const int max_depth = handle_depth_input();
   const int new_vertices_num = handle_new_vertices_num_input();
+
   const auto params = GraphGenerator::Params(max_depth, new_vertices_num);
+  auto graphs = std::vector<Graph>();
   auto generation_controller =
       GraphGenerationController(threads_count, graphs_count, params);
+
   auto& logger = Logger::get_instance();
-  std::filesystem::create_directory("./temp");
-  logger.set_file("./temp/log.txt");
-  auto graphs = std::vector<Graph>();
-  graphs.reserve(graphs_count);
 
+  prepare_temp_directory();
+  prepare_log_file(logger);
+
+  graphs.reserve(graphs_count);
   generation_controller.generate(
       [&logger](int index) { log_start(logger, index); },
       [&logger, &graphs](int index, Graph graph) {
         log_end(logger, graph, index);
         graphs.push_back(graph);
+        // GraphTraversal traversal(graph);
+        // log_traversal_start(logger, index);
+        // log_traversal_end(logger, traversal, index);
         const auto graph_printer = GraphPrinter(graph);
         write_to_file(graph_printer,
                       "./temp/graph_" + std::to_string(index) + ".json");
       });
+
+  // traverse_graphs(graphs, logger);
+
+  auto traversal_controller = GraphTraversalController(graphs);
+  traversal_controller.traverse(
+      [&logger](int index) { log_traversal_start(logger, index); },
+      [&logger, &graphs](int index, std::vector<GraphTraversal::Path> path) {

```

Comment:
Будет логичным возвращать и сам граф тоже
```suggestion
      [&logger](int index, const Graph& graph) { log_traversal_start(logger, index); },
      [&logger, &graphs](int index, const Graph& graph, std::vector<GraphTraversal::Path> path) {
```